cli/mender-artifact/partition.go

```diff
@@ -277,13 +277,20 @@ func (a *artifactExtFile) Close() (err error) {
 	if a == nil {
 		return nil
 	}
-	if a.repack {
-		err = repackArtifact(a.comp, a.name, a.path, "")
-	}
 	if a.tmpf != nil {
+		if a.flush {
+			a.tmpf.Sync()
```

Comment:
Flush and Sync are two different things: Flush outputs data that the application has buffered in memory, Sync forces the kernel cache to be dumped to disk.

`os.File` doesn't need flushing because everything is a direct write (as opposed to `bufio.Writer`, which does have buffering/flushing). And there is definitely no need for Sync here, as it only decreases performance and provides no benefit (the kernel cache will survive even if your program crashes). Sync is only for system level changes where an application needs to recover from spontaneous reboots, and you aren't going to do that here.